flink-clients/src/main/java/org/apache/flink/client/deployment/executors/RemoteExecutor.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.client.deployment.executors;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.core.execution.Executor;
+
+/**
+ * The {@link Executor} to be used when executing a job on an already running cluster.
+ */
+@Internal
+public class RemoteExecutor extends StandaloneSessionClusterExecutor {
```

Comment:
I'm not sure whether or not it would be better to be

```java

public class RemoteExecutor extends AbstractSessionClusterExecutor<StandaloneClusterId, StandaloneClientFactory> {

	public static final String NAME = "remote-cluster";

	public StandaloneSessionClusterExecutor() {
		super(new StandaloneClientFactory());
	}
}
```

where I'm afraid in current diff we possible have

```java
StandaloneSessionClusterExecutor executor = ...;
executor.NAME == ?
```

but let's say I don't stick to it at all because within our discovery mechanism it should not be a problem.

Reply:
Good point @TisonKun ! Thanks for spotting this. I will address that.